### PR TITLE
feat(FAB): Add Lightweight Styling to FloatingActionButton

### PR DESCRIPTION
**Pull request by lesterbotello**
﻿GitHub Issue: [249](https://github.com/unoplatform/uno.themes-private/issues/10)

## PR Type

What kind of change does this PR introduce?

- Feature

## Description

Adds support for Material lightweight styling

## PR Checklist 
Please check if your PR fulfills the following requirements:

- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] ~~Tested UWP~~
- [ ] ~~Tested iOS~~
- [ ] ~~Tested Android~~
- [ ] ~~Tested WASM~~
- [ ] ~~Tested MacOS~~
- [X] Contains **No** breaking changes
  > If the pull request contains breaking changes, commit message must contain a detailed description of the action to take for the consumer of this library. As explained by the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+						[ResourceKeyDefinition(typeof(Brush), "FabTertiaryStateOverlayBackgroundPointerOver")]
+						public static ResourceValue<Brush> PointerOver => new("FabTertiaryStateOverlayBackgroundPointerOver", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabTertiaryStateOverlayBackgroundFocused")]
+						public static ResourceValue<Brush> Focused => new("FabTertiaryStateOverlayBackgroundFocused", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabTertiaryStateOverlayBackgroundDisabled")]
+						public static ResourceValue<Brush> Disabled => new("FabTertiaryStateOverlayBackgroundDisabled", true);
+					}
+				}
```
Comment (agneszitte) [inline src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs:226]: Missing Styles at the end @lesterbotello, example here:
https://github.com/unoplatform/Uno.Themes/blob/d6d4d00ecb8ea8b4834010a5112870431008fb16/src/library/Uno.Themes.WinUI.Markup/Theme/ComboBox.cs#L293

Comment (lesterbotello) [reply src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs:227]: Done

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			}
+
+			public static class Styles
```
Comment (agneszitte) [inline src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs:230]: Missing some styles here @lesterbotello to follow the XAML
- MaterialSurfaceFabStyle
- MaterialSecondaryFabStyle
- MaterialTertiaryFabStyle
- MaterialSmallFabStyle
- MaterialSurfaceSmallFabStyle
- MaterialSecondarySmallFabStyle
- MaterialTertiarySmallFabStyle
- MaterialLargeFabStyle
- MaterialSurfaceLargeFabStyle
- MaterialSecondaryLargeFabStyle
- MaterialTertiaryLargeFabStyle

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+						[ResourceKeyDefinition(typeof(Brush), "FabStateOverlayBackgroundPointerOver")]
+						public static ResourceValue<Brush> PointerOver => new("FabStateOverlayBackgroundPointerOver", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabStateOverlayBackgroundFocused")]
+						public static ResourceValue<Brush> Focused => new("FabStateOverlayBackgroundFocused", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabStateOverlayBackgroundDisabled")]
+						public static ResourceValue<Brush> Disabled => new("FabStateOverlayBackgroundDisabled", true);
+					}
+				}
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs:65]: Should be inside of the `FloatingActionButton.Resources.Default` class

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+						[ResourceKeyDefinition(typeof(Brush), "FabSurfaceStateOverlayBackgroundPointerOver")]
+						public static ResourceValue<Brush> PointerOver => new("FabSurfaceStateOverlayBackgroundPointerOver", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabSurfaceStateOverlayBackgroundFocused")]
+						public static ResourceValue<Brush> Focused => new("FabSurfaceStateOverlayBackgroundFocused", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabSurfaceStateOverlayBackgroundDisabled")]
+						public static ResourceValue<Brush> Disabled => new("FabSurfaceStateOverlayBackgroundDisabled", true);
+					}
+				}
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs:119]: Should be inside of the `FloatingActionButton.Resources.Surface` class and renamed to `StateOverlay`

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+						[ResourceKeyDefinition(typeof(Brush), "FabSecondaryStateOverlayBackgroundPointerOver")]
+						public static ResourceValue<Brush> PointerOver => new("FabSecondaryStateOverlayBackgroundPointerOver", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabSecondaryStateOverlayBackgroundFocused")]
+						public static ResourceValue<Brush> Focused => new("FabSecondaryStateOverlayBackgroundFocused", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabSecondaryStateOverlayBackgroundDisabled")]
+						public static ResourceValue<Brush> Disabled => new("FabSecondaryStateOverlayBackgroundDisabled", true);
+					}
+				}
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs:173]: Should be inside of the `FloatingActionButton.Resources.Secondary` class and renamed to `StateOverlay`

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+						[ResourceKeyDefinition(typeof(Brush), "FabTertiaryStateOverlayBackgroundPointerOver")]
+						public static ResourceValue<Brush> PointerOver => new("FabTertiaryStateOverlayBackgroundPointerOver", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabTertiaryStateOverlayBackgroundFocused")]
+						public static ResourceValue<Brush> Focused => new("FabTertiaryStateOverlayBackgroundFocused", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "FabTertiaryStateOverlayBackgroundDisabled")]
+						public static ResourceValue<Brush> Disabled => new("FabTertiaryStateOverlayBackgroundDisabled", true);
+					}
+				}
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs:227]: Should be inside of the `FloatingActionButton.Resources.Tertiary` class and renamed to `StateOverlay`

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs (hunk 5 above)
Comment (lesterbotello) [reply src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs]: Done

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs (hunk 6 above)
Comment (lesterbotello) [reply src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs]: Done

File: src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs (hunk 3 above)
Comment (lesterbotello) [reply src/library/Uno.Themes.WinUI.Markup/Themes/FloatingActionButton.cs]: Done.

**Review by kazo0 — APPROVED** [review]

**Pull request merged**